framework/src/main/java/org/checkerframework/framework/util/dependenttypes/DependentTypesHelper.java

```diff
@@ -256,198 +272,86 @@ private void viewpointAdaptExecutable(
-            AnnotatedExecutableType methodType,
-            List<? extends ExpressionTree> argTrees) {
-
-        if (tree.getKind() == Kind.METHOD_INVOCATION) {
-            ExecutableElement method = TreeUtils.elementFromUse((MethodInvocationTree) tree);
-            if (isVarArgsInvocation(method, methodType, argTrees)) {
-                List<JavaExpression> result = new ArrayList<>();
-
-                for (int i = 0; i < method.getParameters().size() - 1; i++) {
-                    result.add(JavaExpression.fromTree(argTrees.get(i)));
-                }
-                List<JavaExpression> varargArgs = new ArrayList<>();
-                for (int i = method.getParameters().size() - 1; i < argTrees.size(); i++) {
-                    varargArgs.add(JavaExpression.fromTree(argTrees.get(i)));
-                }
-                Element varargsElement =
-                        method.getParameters().get(method.getParameters().size() - 1);
-                TypeMirror tm = ElementUtils.getType(varargsElement);
-                result.add(new ArrayCreation(tm, Collections.emptyList(), varargArgs));
-
-                return result;
-            }
+    public void atFieldAccess(MemberSelectTree fieldAccess, AnnotatedTypeMirror type) {
+        if (!hasDependentType(type) || TreeUtils.isClassLiteral(fieldAccess)) {
+            return;
         }
 
-        return SystemUtil.mapList(JavaExpression::fromTree, argTrees);
-    }
-
-    /**
-     * Returns true if method is a varargs method or constructor and its varargs arguments are not
-     * passed in an array.
-     *
-     * @param method the method or constructor
-     * @param methodType type of the method or constructor; used for determining the type of the
-     *     varargs formal parameter
-     * @param args the arguments at the call site
-     * @return true if method is a varargs method and its varargs arguments are not passed in an
-     *     array
-     */
-    private boolean isVarArgsInvocation(
-            ExecutableElement method,
-            AnnotatedExecutableType methodType,
-            List<? extends ExpressionTree> args) {
-        if (method != null && method.isVarArgs()) {
-            if (method.getParameters().size() != args.size()) {
-                return true;
-            }
-            AnnotatedTypeMirror lastArg = factory.getAnnotatedType(args.get(args.size() - 1));
-            List<AnnotatedTypeMirror> paramTypes = methodType.getParameterTypes();
-            AnnotatedArrayType lastParam =
-                    (AnnotatedArrayType) paramTypes.get(paramTypes.size() - 1);
-            return lastArg.getKind() != TypeKind.ARRAY
-                    || AnnotatedTypes.getArrayDepth(lastParam)
-                            != AnnotatedTypes.getArrayDepth((AnnotatedArrayType) lastArg);
+        Element ele = TreeUtils.elementFromUse(fieldAccess);
+        if (ele.getKind() != ElementKind.FIELD && ele.getKind() != ElementKind.ENUM_CONSTANT) {
```

Comment:
What else might it be, other than these?  The documentation says it is a field access, but is it actually an arbitrary MemberSelectTree that might not be a field access?